src/planner/plan/append_extend.cpp

```diff
@@ -152,6 +152,13 @@ void Planner::appendRecursiveExtendAsGDS(const std::shared_ptr<NodeExpression>&
         createPathNodeFilterPlan(recursiveInfo->node, recursiveInfo->originalNodePredicate, p);
         gdsCall->ptrCast<LogicalGDSCall>()->setNodePredicateRoot(p.getLastOperator());
     }
+    // E.g. Given schema person-knows->person & person-knows->animal
+    // And query MATCH (a:person:animal)-[e*]->(b:person)
+    // The destination node b after GDS will contain both person & animal label. We need to prune
+    // the animal out.
+    if (nbrNode->getTableIDsSet().size() < recursiveInfo->node->getTableIDsSet().size()) {
```

Comment:
Should we take intersect of the two tableIDsSet here? I'm not sure how the labels are pruned here. but is it possible that there are cases, for example, `recursiveInfo->node` has the tableIDsSet of `person, animal, student`, while `nbrNode`'s tableIDsSet is `person, animal, teacher`?